src/Symfony/Component/Yaml/Tests/ParserTest.php

```diff
@@ -1614,6 +1615,206 @@ public function multiLineDataProvider()
+    'bar': ['baz']
+}
+YAML
+                ,
+            ],
+            'sequence spanning multiple lines nested in mapping' => [
+                [
+                    'foobar' => [
+                        'foo',
+                        'bar',
+                        'baz',
+                    ],
+                ],
+                <<<YAML
+foobar: [foo,
+    bar,
+    baz
+]
+YAML
+                ,
+            ],
+            'nested sequence nested in mapping starting on the same line' => [
+                [
+                    'foo' => [
+                        'foobar',
+                        [
+                            'bar',
+                            'baz',
+                        ],
+                    ],
+                ],
+                <<<YAML
+foo: [foobar, [
+    bar,
+    baz
+]]
+YAML
+                ,
+            ],
+            'nested sequence nested in mapping starting on the following line' => [
+                [
+                    'foo' => [
+                        'foobar',
+                        [
+                            'bar',
+                            'baz',
+                        ],
+                    ],
+                ],
+                <<<YAML
+foo: [foobar,
+    [
+        bar,
+        baz
+]]
+YAML
+                ,
+            ],
+            'mapping nested in sequence' => [
+                ['foo', ['bar' => 'baz']],
+                <<<YAML
+[
+    'foo',
+    {
+        'bar': 'baz'
+    }
+]
+YAML
+                ,
+            ],
+            'mapping spanning multiple lines nested in sequence' => [
+                [
+                    [
+                        'foo' => 'bar',
+                        'bar' => 'baz',
+                    ],
+                ],
+                <<<YAML
+- {
+    foo: bar,
+    bar: baz
+}
+YAML
+                ,
+            ],
+            'nested mapping nested in sequence starting on the same line' => [
+                [
+                    [
+                        'foo' => [
+                            'bar' => 'foobar',
+                        ],
+                        'bar' => 'baz',
+                    ],
+                ],
+                <<<YAML
+- { foo: {
+        bar: foobar
+    },
+    bar: baz
+}
+YAML
+                ,
+            ],
+            'nested mapping nested in sequence starting on the following line' => [
+                [
+                    [
+                        'foo' => [
+                            'bar' => 'foobar',
+                        ],
+                        'bar' => 'baz',
+                    ],
+                ],
+                <<<YAML
+- { foo:
+    {
+        bar: foobar
+    },
+    bar: baz
+}
+YAML
+                ,
+            ],
+            'single quoted multi-line string' => [
+                "foo\nbar",
+                <<<YAML
+'foo
+
+bar'
+YAML
+                ,
+            ],
+            'double quoted multi-line string' => [
```

Comment:
The YAML content of this test doesn't match the description. It should use double quotes, right?